### PR TITLE
Update to JUnit5

### PR DESCRIPTION
**Pull request by parlough**
All tests are passing.

This is just for future maintenance reasons and improved future tests as JUnit4 is no longer updated.

### CODE REVIEW

**Comment by lucko** [comment]
I don't really know enough about JUnit & Gradle to properly review this. Is it even worthwhile updating? Lots of projects still seem to be using v4, is the new version considered stable?

Only thing I noticed was that the imports order has been messed up, static should be at the bottom.

@kashike any opinions?

**Comment by kashike** [comment]
Yes, JUnit 5 is stable - projects should be migrating to it.

File: build.gradle [hunk 1]
```diff
@@ -41,11 +41,17 @@ subprojects {
 
   repositories {
     mavenLocal()
-    mavenCentral()
+    jcenter()
```
Comment (kashike) [inline build.gradle]: Why have you made this change? Central should be in this list no matter what.

File: configurate-core/src/test/java/ninja/leaping/configurate/SimpleConfigurationNodeTest.java [hunk 2]
```diff
@@ -16,11 +16,18 @@
  */
 package ninja.leaping.configurate;
 
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotSame;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
```
Comment (lucko) [inline configurate-core/src/test/java/ninja/leaping/configurate/SimpleConfigurationNodeTest.java]: Static imports should be after all other imports.

I also think it's probably ok to use wildcard import statements in the test classes.

**Comment by parlough** [comment]
Yeah it's been stable for a while and has some nice features. I'll fix the imports and add `mavenCentral()` back soon.

**Pull request merged**